### ISSUE TITLE
Ability to disable on some profile forms

### ISSUE THREAD

**Issue by MtnPavlas** [TechToThePeople/noverwrite#28]
Is there a way to disable this on specific profile forms? I love having this capability enabled on event registration, donation and membership forms, but do want to have a dedicated form available for the users to deliberately update their first and last names. I'm thinking the easiest way might be a piece of javascript/jQuery to be included in the profile or profile field settings? Thanks for any pointers.

**Comment by twomice** [comment]
Yes, it would be nice to support that option somehow. I'm open to a PR but will probably not have time to make that happen on my own very soon. If you make any progress and would like to submit a PR, please do, and I'll be glad to review/comment and/or merge. Thanks @MtnPavlas !

**Comment by MtnPavlas** [comment]
Thanks, I'm not super strong with Civi so not sure how to make this configurable, but for my purposes, I simply updated the file **noverwrite.php**, line 6 from
`$names = array ("CRM_Profile_Form_Edit","CRM_Event_Form_Registration_Register","CRM_Contribute_Form_Contribution_Main");`
to
`$names = array ("CRM_Event_Form_Registration_Register","CRM_Contribute_Form_Contribution_Main");`

The above results in event registration and contribution forms to be protected, but any other forms will not - for our purposes that works better since when someone is editing a straight out profile (i.e., not on a contribution or event registration page), they are doing so deliberately with the objective to update all those fields on the profile.

While not a configurable solution, I hope this helps someone.
